modules/openxr/doc_classes/OpenXRSpatialAnchorCapability.xml

```diff
@@ -0,0 +1,120 @@
+			<param index="0" name="scope" type="int" enum="OpenXRSpatialAnchorCapability.OpenXrSpatialPersistenceScope" />
+			<description>
+				Returns [code]true[/code] if this persistence scope is supported by our spatial anchor capability.
+				[b]Note:[/b] only valid after an OpenXR instance has been created.
+			</description>
+		</method>
+		<method name="is_spatial_anchor_supported">
+			<return type="bool" />
+			<description>
+				Returns [code]true[/code] if spatial anchors is supported by the hardware. Only returns a valid value after OpenXR is initialized.
+			</description>
+		</method>
+		<method name="is_spatial_persistence_supported">
+			<return type="bool" />
+			<description>
+				Returns [code]true[/code] if persistent spatial anchors is supported by the hardware. Only returns a valid value after OpenXR is initialized.
+			</description>
+		</method>
+		<method name="make_anchor_persistent">
+			<return type="bool" />
+			<param index="0" name="anchor_tracker" type="OpenXRAnchorTracker" />
+			<param index="1" name="persistence_context" type="RID" default="RID()" />
+			<param index="2" name="callback" type="Callable" default="Callable()" />
+			<description>
+				Changes this anchor to a persistent anchor. This means its location will be stored on the device and the anchor will be restored the next time your application starts. If [param persistence_context] is not specified the default will be used, this requires [member ProjectSettings.xr/openxr/extensions/spatial_entity/enable_builtin_anchor_detection] to be set.
+				[b]Note:[/b] This is an asynchronous method. This method returns false if we fail to start the method. On success [param callback] will be called. On failure [signal anchor_persistent_failed] will be emitted.
+			</description>
+		</method>
+		<method name="make_anchor_unpersistent">
+			<return type="bool" />
+			<param index="0" name="anchor_tracker" type="OpenXRAnchorTracker" />
+			<param index="1" name="persistence_context" type="RID" default="RID()" />
+			<param index="2" name="callback" type="Callable" default="Callable()" />
+			<description>
+				Remotes the persistent data from this anchor. The runtime will not recreate the anchor when your application restarts. If [param persistence_context] is not specified the default will be used, this requires [member ProjectSettings.xr/openxr/extensions/spatial_entity/enabled] to be set.
+				[b]Note:[/b] This is an asynchronous method. This method returns false if we fail to start the method. On success [param callback] will be called. On failure [signal anchor_unpersistent_failed] will be emitted.
+			</description>
+		</method>
+		<method name="remove_anchor">
+			<return type="void" />
+			<param index="0" name="anchor_tracker" type="OpenXRAnchorTracker" />
+			<description>
+				Remove an anchor previously created with [method create_new_anchor]. If this anchor was persistent you must first call [method make_anchor_unpersistent] and await its callback.
```

Comment:
> If this anchor was persistent you must first call [method make_anchor_unpersistent] and await its callback.

Isn't something we could do automatically?

Reply:
I thought about that but it has problems because making an anchor no longer persistent, is an async method. Especially if removing anchors is part of some cleanup logic, we potentially need to ensure the user delays part of their cleanup.

I'm on the fence though. We could make removing an anchor an asynchronous method on our side, even if it would be immediate if the anchor is not persistent and deal with it that way.